portfolio/src/main/webapp/js/background.cycle.js

```diff
@@ -0,0 +1,157 @@
+var SCALING_MODE_CONTAIN = 3; //Sets 'background-size' to 'contain'. This scales the bakcground image to the largest size such that both its width and its height can fit inside the content area
+
+/**
+ * Adds a cycling (fading) background to the selected element
+ * @param {Object} options Options for tweaking the cycle setings.
+ * imageUrls: an array of strings representing urls to the images to cycle through
+ * duration: the nr of miliseconds between two fades.
+ * fadeSpeed: the nr of miliseconds it takes for one image to fade out to another.
+ * backgroundSize: specify a value for the css3 property 'background size' or one of the following constants; SCALING_MODE_NONE, SCALING_MODE_STRETCH, SCALING_MODE_COVER, SCALING_MODE_CONTAIN
+ */
+$.fn.backgroundCycle = function(options) {
+  var settings = $.extend(
+    {
+      imageUrls: [],
+      duration: 5000,
+      fadeSpeed: 1000,
+      backgroundSize: SCALING_MODE_NONE
+    },
+    options
+  );
+
+  fadeSpeed = settings.fadeSpeed;
+
+  var marginTop = this.css("margin-top");
+  var marginRight = this.css("margin-right");
+  var marginBottom = this.css("margin-bottom");
+  var marginLeft = this.css("margin-left");
+
+  if (!this.is("body")) {
+    this.css({
+      position: "relative"
+    });
+  }
+
+  var contents = $(document.createElement("div"));
+
+  var children = this.children().detach();
+  contents.append(children);
+
+  imageIds = new Array();
+
+  for (var i = 0; i < settings.imageUrls.length; i++) {
+    var id = "bgImage" + i;
+    var src = settings.imageUrls[i];
+    var cssClass = "cycle-bg-image";
+
+    var image = $(document.createElement("div"));
+    image.attr("id", id);
+    image.attr("class", cssClass);
+
+    var sizeMode;
+
+    switch (settings.backgroundSize) {
+      default:
+        sizeMode = settings.backgroundSize;
+        break;
+      case SCALING_MODE_NONE:
+        sizeMode = "auto";
+        break;
+      case SCALING_MODE_STRETCH:
+        sizeMode = "100% 100%";
+        break;
+      case SCALING_MODE_COVER:
+        sizeMode = "cover";
+        break;
+      case SCALING_MODE_CONTAIN:
+        sizeMode = "contain";
+        break;
+    }
+
+    image.css({
+      "background-image": "url('" + src + "')",
+      "background-repeat": "no-repeat",
+      "background-position": "bottom",
+      "background-size": sizeMode,
+      "-moz-background-size": sizeMode,
+      "-webkit-background-size": sizeMode,
+      position: "fixed",
+      left: marginLeft,
+      top: marginTop,
+      right: marginRight,
+      bottom: marginBottom
+    });
+
+    this.append(image);
+
+    imageIds.push(id);
+  }
+
+  contents.css({
+    position: "absolute",
+    left: marginLeft,
+    top: marginTop,
+    right: marginRight,
+    bottom: marginBottom
+  });
+
+  this.append(contents);
+  $(".cycle-bg-image").hide();
+  $("#" + imageIds[0]).fadeIn();
+
+  if (settings.duration > 0) {
+    setInterval(cycleToNextImage, settings.duration);
+  }
+
+  return {
```

Comment:
do you need to return this function here, or just need to define an inner function?